server/sonar-server/src/test/java/org/sonar/server/computation/step/PersistTestsStepTest.java

```diff
@@ -0,0 +1,284 @@
+  private static final String MAIN_FILE_UUID_2 = "MAIN-FILE-2";
+
+  PersistTestsStep sut;
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+
+  @ClassRule
+  public static DbTester db = new DbTester();
+
+  File reportDir;
+  DbSession session;
+  DbClient dbClient;
+  System2 system2;
+
+  long now = 123456789L;
+
+  @Before
+  public void setup() throws Exception {
+    db.truncateTables();
+    session = db.myBatis().openSession(false);
+    dbClient = new DbClient(db.database(), db.myBatis(), new FileSourceDao(db.myBatis()));
+    reportDir = temp.newFolder();
+
+    system2 = mock(System2.class);
+    when(system2.now()).thenReturn(now);
+    sut = new PersistTestsStep(dbClient, system2);
+
+    initBasicReport();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    MyBatis.closeQuietly(session);
+
+  }
+
+  @Override
+  protected ComputationStep step() throws IOException {
+    return sut;
+  }
+
+  @Test
+  public void no_test_in_database_and_batch_report() throws Exception {
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), ComponentTesting.newProjectDto()));
+
+    assertThat(dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1)).isNull();
+  }
+
+  @Test
+  public void insert_tests_in_report() throws Exception {
+    BatchReportWriter writer = new BatchReportWriter(reportDir);
+    List<BatchReport.Test> batchTests = Arrays.asList(
+      newTest(1), newTest(2)
+      );
+    writer.writeTests(TEST_FILE_REF_1, batchTests);
+    List<CoverageDetail> coverageDetails = Arrays.asList(
+      newCoverageDetail(1, MAIN_FILE_REF_1)
+      );
+    writer.writeCoverageDetails(TEST_FILE_REF_1, coverageDetails);
+
+    sut.execute(new ComputationContext(new BatchReportReader(reportDir), ComponentTesting.newProjectDto(PROJECT_UUID)));
+
+    FileSourceDto dto = dbClient.fileSourceDao().selectTest(TEST_FILE_UUID_1);
+    assertThat(dto.getCreatedAt()).isEqualTo(now);
+    assertThat(dto.getUpdatedAt()).isEqualTo(now);
+    assertThat(dto.getProjectUuid()).isEqualTo(PROJECT_UUID);
+    assertThat(dto.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    assertThat(dto.getTestData()).hasSize(2);
+    FileSourceDb.Test test1 = dto.getTestData().get(0);
+    assertThat(test1.getFileUuid()).isEqualTo(TEST_FILE_UUID_1);
+    assertThat(test1.getName()).isEqualTo("name#1");
+    assertThat(test1.getMsg()).isEqualTo("message#1");
+    assertThat(test1.getStacktrace()).isEqualTo("stacktrace#1");
+    assertThat(test1.getStatus()).isEqualTo(TestStatus.FAILURE);
+    assertThat(test1.getType()).isEqualTo(TestType.UT);
+    assertThat(test1.getExecutionTimeMs()).isEqualTo(1_000);
+    assertThat(test1.getCoveredFileCount()).isEqualTo(1);
+    assertThat(test1.getCoveredFile(0).getCoveredLineList()).containsExactly(1, 2, 3);
+    assertThat(test1.getCoveredFile(0).getFileUuid()).isEqualTo(MAIN_FILE_UUID_1);
+    FileSourceDb.Test test2 = dto.getTestData().get(1);
```

Comment:
I would separate this use case into another test "insert_minimal_data_tests_in_report" (or another name :) )